### DISCUSSION TITLE
Quasar fresh install is broken after install any vue module

### DISCUSSION THREAD

**Discussion by leostereo** [quasarframework/quasar#10110] [General - Components / Directives / etc] [answered]
Hello , to reproduce this issue Im doing:
quasar create (to create fresh project)
quasar dev to run the project.
Then , in another console I try installing some vue module , like:
`npm install vue-particles --save-dev
`
But , then , quasar stop compiling with errors , I got:

```

 App •  ERROR  •  UI  in ./node_modules/core-js/modules/es.array.flat.js

Error: no such file or directory, open 'C:\Users\ingro\Documents\cuyolabs\web\node_modules\core-js\modules\es.array.flat.js'


 App •  ERROR  •  UI  in ./node_modules/core-js/modules/es.regexp.constructor.js

Error: no such file or directory, open 'C:\Users\ingro\Documents\cuyolabs\web\node_modules\core-js\modules\es.regexp.constructor.js'


 App •  ERROR  •  UI  in ./node_modules/core-js/modules/es.regexp.exec.js

Error: no such file or directory, open 'C:\Users\ingro\Documents\cuyolabs\web\node_modules\core-js\modules\es.regexp.exec.js'


 App •  ERROR  •  UI  in ./node_modules/core-js/modules/es.string.replace.js

Error: no such file or directory, open 'C:\Users\ingro\Documents\cuyolabs\web\node_modules\core-js\modules\es.string.replace.js'
```
As if some components where deleted ...
Very strange , never happened this to me before.
Also tryed with other modules , and have same result.
Hope someone can help me ...
Just wanted to learn about boot files when founding this behaviour.

Leandro.

**Comment by ontwikkelfabriek** [comment]
@leostereo 

Probable nothing to do with your problem but:

I would never put any of my Quasar ( nodemodules npm) projects into a deep subfolder as the Windows user documents folder. That's because In Windows, there is a limit for 260 characters in any path. This includes filenames, so a file cannot have more characters than 260-directory path length.

This means that you could have quite a lot of subdirectories, but as you go deeper, the maximum filename gets shorter. This can and will give `problems` with deep nested `node modules`.

Instead I put my projects in something like 'D:\projects`

**Reply by leostereo** [reply]
Thanks @ontwikkelfabriek  ; I was not  aware on that.
I tryed with a fresh project located on a root directory but still have same problem.
I though mt problem was related to mixing npm and yarn , so I created a fresh project, choosing yarn as packet manager.
Then I did:

```
D:\quasar\web>yarn add particles.vue vue@2
yarn add v1.22.5
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
info fsevents@2.3.2: The platform "win32" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning " > @babel/eslint-parser@7.14.7" has unmet peer dependency "@babel/core@>=7.11.0".
warning " > eslint-webpack-plugin@2.5.4" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
[5/5] Building fresh packages...
success Saved lockfile.
success Saved 6 new dependencies.
info Direct dependencies
├─ particles.vue@2.14.2
└─ vue@2.6.14
info All dependencies
├─ particles.vue@2.14.2
├─ pathseg@1.2.0
├─ tsparticles@1.31.1
├─ vue-class-component@7.2.6
├─ vue-property-decorator@9.1.2
└─ vue@2.6.14
Done in 13.02s.
```

And , in my  "quasar dev" console , I got:

```
 App •  WAIT  • Compiling of "UI" in progress...
 App •  DONE  • "UI" compiled with errors • 225ms



 App •  ERROR  •  UI  in ./node_modules/vue-router/dist/vue-router.esm-bundler.js

Error: no such file or directory, open 'D:\quasar\web\node_modules\vue-router\dist\vue-router.esm-bundler.js'



 App •  COMPILATION FAILED  • Please check the log above for details.

 App •  WAIT  • Compiling of "UI" in progress...

 App •  DONE  • "UI" compiled with errors • 2681ms



 App •  ERROR  •  UI  in ./node_modules/vue-router/dist/vue-router.esm-bundler.js

Error: no such file or directory, open 'D:\quasar\web\node_modules\vue-router\dist\vue-router.esm-bundler.js'



 App •  COMPILATION FAILED  • Please check the log above for details.


```
I will make some other tests ... 
Thanks.

**Reply by leostereo** [reply]
mmm not ... any package I attempt to install make "quasar dev" to crash.
I tryed with
 npm install vuejs-datepicker --save
And have the same result.
Can be something wrong with my quasar installation ? 
Im using:
 quasar - 2.0.3 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 3.0.3 -- Quasar Framework local CLI
  @quasar/extras - 1.10.8 -- Quasar Framework fonts, icons and animations
  
  Any ideas ? 
  Regards.

**Reply by MilosPaunovic** [reply]
Just to be on the safe side, have you tried to delete `node_modules` and `package-lock.json` and do a fresh `npm i`?

**Reply by hawkeye64** [reply]
Are you installing while "quasar dev" is running? If so, never a good idea...

**Reply by leostereo** [reply]
> Are you installing while "quasar dev" is running? If so, never a good idea...

Yes ... I was ... sorry didnt know it was not good idea, thanks.

**Reply by leostereo** [reply]
> Just to be on the safe side, have you tried to delete `node_modules` and `package-lock.json` and do a fresh `npm i`?

I will try and let you know how it goes.

**Reply by leostereo** [reply]
> Are you installing while "quasar dev" is running? If so, never a good idea...

Im , running a fresh install ... and it  keeps happening , can not install any single library, very frustrating.

**Comment by hawkeye64** [comment, marked as answer]
I just noticed this:
```
yarn add particles.vue vue@2
```
So, to me, it looks like you are using Quasar v2 (which uses Vue 3), but then you are trying to use vue@2 and particles.vue (which is also vue@2). Meaning you are mixing different versions of Vue.

If you want to use particles.vue, you will have to use Quasar v1
```
quasar create myTestApp --branch v1
```
Don't install Vue v2 as it is already installed under the hood. Install particles.vue. You will need to write a boot file for it. https://v1.quasar.dev/quasar-cli/boot-files#introduction

**Reply by matteobruni** [reply]
Or install `particles.vue3` that works with Vue3 😏

**Reply by leostereo** [reply]
I can finally close this discussion.
Thanks you all .... I was mixing everithing.
I finally was able to install particles and other modules.
